modules/http.lua

```diff
@@ -222,9 +230,21 @@ function ClientRequest:initialize(options, callback)
     if cfound then
       connection_found = value
     end
+    local uafound = key:lower() == 'user-agent'
+    if uafound then
+      user_agent_found = value
+    end
     table.insert(self, header)
   end
 
+  if not user_agent then
+    user_agent = self.getDefaultUserAgent()
+  end
+
+  if user_agent ~= '' then
```

Comment:
it does always put it unless there is one in the headers provided by the user already